openpype/pipeline/file_collection.py

```diff
@@ -0,0 +1,274 @@
+    reviewable=False,
+):
+    """Get representation with expected files.
+
+    Args:
+        instance (pyblish.Instance): instance
+        file_path (str): file path
+        frame_start (Optional[int]): first frame
+        frame_end (Optional[int]): last frame
+        log (Optional[Logger]): logger
+        only_existing (Optional[bool]): Ensure that files exists.
+        reviewable (Optional[bool]): reviewable
+
+
+    Returns:
+        dict[str, Any]: representation
+    """
+    log = log or Logger().get_logger(__name__)
+    file_ext = os.path.splitext(file_path)[-1].lstrip(".")
+    output_dir = os.path.dirname(file_path)
+    # get families from instance and add family from data
+    families = set(
+        instance.data.get("families", []) + [instance.data["family"]]
+    )
+
+    tags = []
+    if reviewable:
+        tags.append("review")
+
+    # prepare base representation data
+    representation = {
+        "name": file_ext,
+        "ext": file_ext,
+        "stagingDir": output_dir,
+        # QUESTION: should we use persistent staging dir always?
+        "stagingDir_persistent": True,
+        "tags": tags,
+    }
+
+    # add frame range data
+    if frame_start is not None:
+        representation["frameStart"] = frame_start
+    if frame_end is not None:
+        representation["frameEnd"] = frame_end
+
+    # collect files from sequential path or single file in list
+    collected_file_frames = \
+        collect_basenames_from_sequential_path(
+            file_path, frame_start, frame_end, only_existing)
+
+    # set slate frame
+    if (
+        "slate" in families
+        # make sure frame range is set
+        and _add_slate_frame_to_collected_frames(
+            collected_file_frames, frame_start, frame_end)
+    ):
+        log.info("Slate frame added to collected frames.")
+        representation["frameStart"] = frame_start - 1
+
+    if len(collected_file_frames) == 1:
+        representation["files"] = collected_file_frames.pop()
+    else:
+        representation["files"] = collected_file_frames
+
+    return representation
+
+
+def _add_slate_frame_to_collected_frames(
+    collected_file_frames,
+    frame_start=None,
+    frame_end=None,
+):
+    """Add slate frame to collected frames.
+
+    Args:
+        collected_file_frames (list[str]): collected file frames
+        frame_start (Optional[int]): first frame
+        frame_end (Optional[int]): last frame
+
+    Returns:
+        Bool: True if slate frame was added
+    """
+    if (
+        frame_start is None
+        and frame_end is None
+    ):
+        return False
+
+    frame_start_str = frames.get_frame_start_str(
+        frame_start, frame_end)
+    frame_length = int(frame_end - frame_start + 1)
+
+    # add slate frame only if it is not already in collected frames
+    if frame_length == len(collected_file_frames):
+        frame_slate_str = frames.get_frame_start_str(
+            frame_start - 1,
+            frame_end
+        )
```

Comment:
The logic here is wrong. You want to insert a file into an existing sequence that already has 'some sort of potential padding' and want to insert a frame before the first.

The `get_frame_start_str` call returns the first frame padded with the length of the last frame.
If the current collected file frames range are
```
frame.999.exr
frame.1000.exr
frame.1001.exr
```
You are now suddenly prepending `frame.0998.exr`
You should actually be matching the padding from the actual padding in the existing sequence for this function to be correct.